lualib/resty/kong/tls.lua

```diff
@@ -99,6 +101,25 @@ if ngx.config.subsystem == "http" then
     end
 
 
+    function _M.set_client_ca_list(ca_list)
+        if get_phase() ~= 'ssl_cert' then
+            error("API disabled in the current context")
+        end
+
+        local r = get_request()
+        if not r then
```

Comment:
I know we already have the discussion about checking `r`,
but the other functions still don't do the same thing, 
Should we change the others to keep consistent？

Reply:
my opinion is to add this check to existing functions as well, although it's indeed redundent
given our phase check. but it gives us extra protection because this a private API from openresty,
it doesn't guaranteeded to be unchanged in the future.

Reply:
I think we have gotten the agreement. Let's do it in another PR, keep this PR clean.